packages/core/src/ceramic.ts

```diff
@@ -302,6 +302,7 @@ export class Ceramic implements CeramicApi {
       this.repository.index,
       this._logger
     )
+    this.repository.index.setSyncQueryApi(this.syncApi)
```

Comment:
There is a way to get rid of the circular dependency: 

`SyncApi` only uses the indexApi to retrieve the indexable models. We can pass the indexable models in `SyncApi.init()`

Reply:
ooh good call.  We'd also have to update it any time the set of indexed models changes, but that should be doable and would indeed resolve the circular dependency.  Probably fine to leave for a follow-up though?

Reply:
That actually only removes one of the circular dependencies. The main one is from 

`this.repository.stateManager.handleUpdate.bind(this.repository.stateManager),`

Since you have to setDeps on repository, which creates a stateManager, which needs a local index. And the local index would need a sync api.